### PR TITLE
Add user token feature

### PR DESCRIPTION
**Pull request by travelist**

### CODE REVIEW

**Comment by travelist** [comment]
@kosukemori I've just aware that you have PR to this repository: https://github.com/CiscoZeus/java-zeusclient/pull/7/files

Please review this PR and i will check your code as well.

File: README.md [hunk 1]
```diff
@@ -20,12 +20,21 @@ mvn -q exec:java -Dexec.mainClass="com.cisco.zeus.ZeusSampleClient"
...
+```
+
 ```java
-import com.cisco.zeus.*;
-ZeusAPIClient zeusClient = new ZeusAPIClient("your_token_here");
+// For BucketName Header
+zeusClient.bucket("organization-name/bucket-name").Methods()
 ```
+
+###Metrics
```
Comment (kosukemori) [inline README.md]: It seems like there are wrong Markdown? (Even before @travelist  modified).
ex) ###Metrics  -> ### Metrics

File: src/main/java/com/cisco/zeus/ZeusAPIClient.java [hunk 2]
```diff
@@ -1,237 +1,228 @@
...
+                throw new IOException(String.format("Unknown Method %s", method));
+        }
 
-        String url = ZEUS_API + path;
+        request.setHeader("Authorization", String.format("Bearer %s", accessToken));
 
-        URL obj = new URL(url);
+        if (bucketName != null) {
+            request.setHeader("Bucket-Name", bucketName);
+            bucketName = null;
```
Comment (kosukemori) [inline src/main/java/com/cisco/zeus/ZeusAPIClient.java:184]: Nice!

**Review by kosukemori — COMMENTED** [review]
Should we deal with old usage (not calling 'bucket') of client?

I think old users can use old client because we still support old external_token in zeus_api & fluentd-server.

So I think we should deprecate external_token from zeusclients.

**Comment by travelist** [comment]
@kosukemori Agreed that we should remove deprecated usage. Which do you think it's better that removing them now or near the future once UI and other services cope with this change (at the moment, user can still make external token)?

**Comment by kosukemori** [comment]
@travelist 
Ideally, I want remove old usage now, and merge to develop branch.
I believe putting "For production zeus, please use this release (or master branch?) at this moment" in top of README is enough for customer.

**Comment by travelist** [comment]
@kosukemori I think it's not now because (1) I think normally users don't check github pages, (2) we have possibility to refactor API endpoints https://github.com/CiscoZeus/zeus_api/issues/148 (please comment on it if you have any ideas) and (3) we should keep backward compatibility unless we completely have removed it.

**Comment by kosukemori** [comment]
@travelist 
Sure, these deprecation depends a project's proceedings. So then let's keep compatibility at once.

Probably I should change go-client's PR to keep compatibility.

**Comment by travelist** [comment]
I guess it's not necessary since the changeability. and agreed that we need to sort out them at some point of time:)

**Review by tokwii — APPROVED** [review]
LGTM

**Pull request merged**